content/en/docs/concepts/overview/working-with-objects/conditions.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Conditions
+content_type: concept
+weight: 30
+---
+
+<!-- overview -->
+In Kubernetes, conditions as a concept are used to provide a detailed and structured status of an object.
+This granular status information helps in understanding status of various Kubernetes objects and facilitates communication between different components of the system.
+
+<!-- body -->
+## Understanding conditions
+
+A condition in Kubernetes includes several key fields:
+
+- **Type**: The specific kind of condition(e.g., `Ready`, `PodScheduled`).
+- **Status**: The state of the condition, which can be `True`, `False`, or `Unknown`.
+- **Reason**: String explaining why the condition is in a current state
+- **Message**: Message explaining details of the message
+- **Last Transition Time**: This represents the timestamp where condition last changed status
+
+## Example: Pod conditions
+
+The Pod API uses conditions to indicate Pod state. Here is an example:
+
+```yaml
+status:
+  conditions:
+  - type: PodScheduled
+    status: "True"
+    lastTransitionTime: "2023-05-10T16:29:24Z"
+  - type: Initialized
+    status: "True"
+    lastTransitionTime: "2023-05-10T16:29:24Z"
+  - type: ContainersReady
+    status: "True"
+    lastTransitionTime: "2023-05-10T16:29:31Z"
+  - type: Ready
+    status: "True"
+    lastTransitionTime: "2023-05-10T16:29:31Z"
+```
+
+In this example, each condition provides specific information about the pod's status, such as whether it has been scheduled, initialized, or is ready to handle traffic.
+
+In example above, each condition provides information as to the status of the pod. Whether the pod has been scheduled, initialized and ready to accept different traffic. Also whether the container in the pod are ready to be started.
+
+## Purpose and usage of conditions
+
+Conditions being an integral part of Kubernetes play very important role by:
+
+* **Status Information**: Conditions provide an in-depth understanding of the existing cluster state or resources. They help the user or a machine understand what’s going on.
+
+* **Communication**: Conditions act as a form of communication inside the cluster. It helps various components like the kubelet, the scheduler talk to each other or custom controllers to send and receive messages.
+
+* **Activity Engine**: They help custom controllers to perform actions. For given conditions, controllers can make HTTP requests, call to cloud API, etc. Once a pod is unschedulable, a controller might create more resources, Websites, etc.
+
+* **Events**:
+
+* They are used to record a single one-time event.
+
+* **Examples**: Kubernetes uses events to report changes during the control plane’s request processing. It uses events for purely informational purposes, trying to help understand what’s going on in the cluster.
```

Comment:
Conditions help to convey machine-readable status about objects, that a human
is also likely to be able to understand. Conditions help
{{< glossary_tooltip term_id="controller" text="controllers" >}} to send and receive
information about the actual state of the cluster. The messaging is indirect (the
sending controller does not have to be aware of which controllers or people might
read that information).

After observing particular conditions, controllers can make HTTP requests (either to
the Kubernetes API, or perhaps to an external cloud provider). Although HTTP integration
is a common pattern, a controller that relies on conditions is free to use any other means
to perform work.

### Example: controller that manages Nodes {#example-node-autoscaling}

After observing that a Pod is unschedulable, a controller acting as a
[node autoscaler](/docs/concepts/cluster-administration/cluster-autoscaling/)
might trigger creation of a new virtual machine that will become a node and
provide additional capacity.

### Example: thermostat {#example-thermostat}

(To illustrate the concept, here is an unlikely but simple example). You could make a
thermostat based on the Kubernetes API, defining a custom API kind named Room.

In this example, when a Room object has the condition `TooCold` become true, a
controller turns on heating.
If the `TooWarm` condition becomes true, the controller activates cooling.
```